app/scripts/services/Message.js

```diff
@@ -0,0 +1,19 @@
+(function() {
+    function Message($firebaseArray) {
+        var ref = firebase.database().ref().child('messages');
+        var messages = $firebaseArray(ref);
+        
+        
+        return {
+            getByRoomId: function (roomId) {
+                var query = ref.orderByChild('roomId').equalTo(roomId);
+                var currentList = $firebaseArray(query);
```

Comment:
Breaking it up a bit tends to make it a bit more readable, and allows you to debug by `console.log` different parts